### PR TITLE
fix: input addr resolver dependency

### PR DESCRIPTION
**Pull request by mkazlauskas**
# Context

KeyAgent.signTransaction takes in an [inputAddressResolver](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/wallet/src/KeyManagement/types.ts#L124) which is passed as an argument in a remote function call, which isn't supported in our web-extension messaging implementation.

# Proposed Solution

Refactor KeyAgent interface: inject `InputResolver` in the constructor of all concrete implementations and remove it as a prop in `KeyAgent.signTransaction` 702dd1a510eff160e702536f2d2778291239ac3d.

As a result, creating and restoring any KeyAgent now requires `{inputResolver}` as 2nd argument. This introduced a circular dependency (Wallet->KeyAgent->InputResolver(WalletUtil)->Wallet) when using `ObservableWallet` as source for the WalletUtil (which is an InputResolver). To overcome this, created a lazy version of WalletUtil 2eca4fc369b4e2baae6e6437c5c749d09853fdd8. To encapsulate the new not-so-obvious initialization logic, exported a new `setupWallet` util.

# Important Changes Introduced

- Add web-extension e2e test to build and sign a transaction that reproduces the problem described in Context
- Unrelated to context of this PR:
  - [test(e2e): fix base tsconfig path for web-extension](https://github.com/input-output-hk/cardano-js-sdk/commit/7be6bf31b80ccecd24f16a19cd7a893f99fdc31a)
  - [test(e2e): rename web-extension create keyagent button](https://github.com/input-output-hk/cardano-js-sdk/pull/332/commits/c4ce8c2e3f9c58e7064a5007260d1059617f1f70)[test(e2e): rename web-extension create keyagent button](https://github.com/input-output-hk/cardano-js-sdk/pull/332/commits/c4ce8c2e3f9c58e7064a5007260d1059617f1f70) 

## Notes

- ~~`setupWallet` has no tests~~ aced5bbb779a4978e48a8b68c99a62de8c16503b
- Tests I **didn't run** after making changes in this PR:
  - `wallet`: `test:hw -t 'Trezor'`
  - `e2e`: `test:blockfrost`, `test:faucet`, `test:cardano-services

### CODE REVIEW

File: packages/e2e/test/web-extension/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../../../tsconfig.json",
+  "extends": "../../../../tsconfig.json",
```
Comment (rhyslbw) [inline packages/e2e/test/web-extension/tsconfig.json:2]: :eyes:

File: packages/wallet/src/services/WalletUtil.ts [hunk 2]
```diff
@@ -104,3 +109,33 @@ export const createWalletUtil = (context: WalletUtilContext) => ({
 });
 
 export type WalletUtil = ReturnType<typeof createWalletUtil>;
+
+type SetWalletUtilContext = (context: WalletUtilContext) => void;
+
+export const createLazyWalletUtil = (): WalletUtil & { initialize: SetWalletUtilContext } => {
```
Comment (rhyslbw) [inline packages/wallet/src/services/WalletUtil.ts:115]: Let's add some docs here

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Looks good. Just a docs request

File: packages/wallet/src/services/WalletUtil.ts (hunk 2 above)
Comment (mkazlauskas) [reply packages/wallet/src/services/WalletUtil.ts:121]: Done in 426da027e260bcbd9bb6608b2438b15df6698450

**Review by rhyslbw — APPROVED** [review]
Nice work @mkazlauskas

**Review by AngelCastilloB — DISMISSED** [review]
Looks great Martynas! :)

**Review by AngelCastilloB — APPROVED** [review]

**Pull request merged**